### ISSUE TITLE
Add streaming support

### ISSUE THREAD

**Issue by samuell** [samuell/scipipe#12]
Should probably be choosable on each output, whether it should stream its output or not! 

... either in the commandline pattern, or as a struct map field.

**Comment by samuell** [comment]
Nope, we'll use `mkfifo` I think.

**Comment by samuell** [comment]
Have to create an "input and parameter reader", as a generator, so that the main loop in tasks can be:

```go
for inTarget, Params := range readInputsAndParams() {
    ...
}
```

**Comment by samuell** [comment]
Implemented, in a few commits, including this one: 94a7118d10467e5421bb11b57ce76ad92a497e62
